webapp/tests/karma/ts/services/analytics-modules.service.spec.ts

```diff
@@ -0,0 +1,119 @@
+import { TestBed } from '@angular/core/testing';
+import sinon from 'sinon';
+import { expect, assert } from 'chai';
+
+import { ScheduledFormsService } from '@mm-services/scheduled-forms.service';
+import { AnalyticsModulesService } from '@mm-services/analytics-modules.service';
+import { AuthService } from '@mm-services/auth.service';
+import { SettingsService } from '@mm-services/settings.service';
+
+describe('AnalyticsModulesService', () => {
+  let service: AnalyticsModulesService;
+  let authService;
+  let scheduledFormsService;
+  let settingsService;
+
+  beforeEach(() => {
+    authService = { has: sinon.stub() };
+    scheduledFormsService = { get: sinon.stub() };
+    settingsService = { get: sinon.stub() };
+
+    TestBed.configureTestingModule({
+      providers: [
+        { provide: AuthService, useValue: authService },
+        { provide: ScheduledFormsService, useValue: scheduledFormsService },
+        { provide: SettingsService, useValue: settingsService },
+      ]
+    });
+
+    service = TestBed.inject(AnalyticsModulesService);
+    authService = TestBed.inject(AuthService);
+    scheduledFormsService = TestBed.inject(ScheduledFormsService);
+    settingsService = TestBed.inject(SettingsService);
```

Comment:
I don't think you need to change the value for these.

Reply:
fixed!